packages/xpetra/sup/Utils/Xpetra_MatrixMatrix.hpp

```diff
@@ -622,7 +577,13 @@ Note: this class is not in the Xpetra_UseShortNames.hpp
           const Tpetra::CrsMatrix<SC,LO,GO,NO>& tpB = Xpetra::Helpers<SC,LO,GO,NO>::Op2TpetraCrs(B);
           RCP<Tpetra::CrsMatrix<SC,LO,GO,NO> >  tpC = Xpetra::Helpers<SC,LO,GO,NO>::Op2NonConstTpetraCrs(C);
 
-          Tpetra::MatrixMatrix::Add(tpA, transposeA, alpha, tpB, transposeB, beta, tpC);
+          RCP<ParameterList> params = rcp(new ParameterList(""));
+          // params->set("Call fillComplete",false);
+          Tpetra::MatrixMatrix::add(alpha, transposeA, tpA, beta, transposeB, tpB, *tpC,
+                                    Xpetra::toTpetra(transposeA ? A.getRangeMap() : A.getDomainMap()),
+                                    Xpetra::toTpetra(transposeA ? A.getDomainMap() : A.getRangeMap()),
+                                    params);
+          C->resumeFill();
```

Comment:
I'd rather not resume fill here, but tell Tpetra to not fill-complete the matrix. That bugs out, however.

Reply:
@cgcgcg You're right, Tpetra didn't behave correctly if "Call fillComplete" is set to false in the parameter list. The documentation for add() promises that params is interpreted the same way RowMatrix::add() does. This is an easy fix, can I just send you a patch to include in this PR?

Reply:
Sure.